### DISCUSSION TITLE
next.config.js redirects causing Uncaught SyntaxError: Unexpected token '<'

### DISCUSSION THREAD

**Discussion by alex-r89** [vercel/next.js#24683] [Help] [answered]
Hi,

I am finding that when I use a redirect inside my `next.config.js`, an error is thrown in the browser console, `Uncaught SyntaxError: Unexpected token '<'`.

I have tested a number of different methods of redirecting, but in its most basic form I have this:
```js
// next.config.js 
module.exports = {
  async redirects() {
    return [{ source: '/((?!maintenance).*)', destination: '/maintenance', permanent: false }]
  }
}
```

Although the redirect seems to work to a degree, this causes the error in the browser console and I was wondering why this is?

Just to note, I tried adding the following code inside a pages `getStaticProps`:
```js
export async function getStaticProps() {
  if (process.env.MAINTENANCE_MODE) {
    return {
      redirect: {
        destination: '/maintenance',
        permanent: false
      }
    }
}
.........
}
```

which worked perfectly, however I dont really want to add this to my 20+ pages. If I could get the redirect working inside `next.config.js`, Id save on a lot of duplicate code.

**Comment by bobarros** [comment, marked as answer]
It's not properly a bug. By paying attention to what you did, you might see you're redirecting everything. 
All requests inside your page it's going to be redirected, like scripts, images, etc.

That's why you're getting errors. All these lines will throw the error that you are mentioning in some clean page, for example:

```html
    <script nomodule="" src="/_next/static/chunks/polyfills.js?ts=1619886952614"></script>
    <script src="/_next/static/chunks/webpack.js?ts=1619886952614"></script>
    <script src="/_next/static/chunks/main.js?ts=1619886952614"></script>
    <script src="/_next/static/chunks/pages/_app.js?ts=1619886952614"></script>
    <script src="/_next/static/chunks/pages/maintenance.js?ts=1619886952614"></script>
    <script src="/_next/static/development/_buildManifest.js?ts=1619886952614"></script>
    <script src="/_next/static/development/_ssgManifest.js?ts=1619886952614"></script>
```

I see two ways that I would take it.
One it's to exclude everything that you need for '/maintenance' to work. A clean HTML would work like so:

```javascript
async redirects() {
    return [{ 
      source: '/((?!maintenance|_next).*)', 
      destination: '/maintenance', 
      permanent: false }]
  }
```

If you need more paths, you have to add to maintenance|_next|img|video ...

Or you can build some static HTML file an try this:

``` javascript
async redirects() {
    return [{ 
      source: '/((?!maintenance).*)', 
      destination: '/maintenance.html', 
      permanent: false }]
}
```

And then you need a maintenance.html file inside public:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    (...)
    <title>Maintanence</title>
  </head>
  <body>(...)</body>
</html>
```

It's going to work because you're out of Next.js context now, and all redirects will not work.

I hope I was clear enough.
All the best.

**Reply by alex-r89** [reply]
Thanks so much @bobarros the first solution worked perfectly and fixed the issues!
